pypose/lietensor/lietensor.py

```diff
@@ -561,8 +601,24 @@ def scale(self, input):
     def identity(cls, *size, **kwargs):
         return Sim3_type.Log(Sim3_type.identity(*size, **kwargs))
 
-    def randn(self, *size, sigma=1, requires_grad=False, **kwargs):
-        data = super().randn(*size, sigma=sigma, **kwargs).detach()
+    def randn(self, *size:_size, sigma=1.0, requires_grad=False, **kwargs):
+        if not isinstance(sigma, collections.abc.Iterable):
+            sigma = _penta(sigma)
+        elif len(sigma)==3:
+            rotation_sigma = _single(sigma[-2])
+            scale_sigma = _single(sigma[-1])
+            translation_sigma = _triple(sigma[0])
+            sigma = translation_sigma+rotation_sigma+scale_sigma
+        else:
+            assert len(sigma)==5
+        size = to_tuple(size)
+        rotation_data = so3_type.randn(*size, sigma=sigma[-2], **kwargs).detach().tensor()
+        scale_data = sigma[-1] * torch.randn(*(size + torch.Size([1])), **kwargs)
+        translation_x = sigma[0] * torch.randn(*(size + torch.Size([1])), **kwargs)
+        translation_y = sigma[1] * torch.randn(*(size + torch.Size([1])), **kwargs)
+        translation_z = sigma[2] * torch.randn(*(size + torch.Size([1])), **kwargs)
+        translation_data = torch.cat([translation_x, translation_y, translation_z], dim=-1)
```

Comment:
change this like the previous ones..., define`translation` instead of `translation_data`